### PR TITLE
pfblockerng: drop the parse_url host-pick branch from the DNSBL ';' step

### LINKED ISSUE #1237: DNSBL: dead parse_url host-pick branch in pfb_dnsbl_extract_host (unreachable after '/' truncation)
Surfaced during the #1119 host-munging extraction (PR #1236) and confirmed by the PR's adversarial review.

`pfb_dnsbl_extract_host()` (`src/usr/local/pkg/pfblockerng/pfblockerng.inc`) contains, in its `;`-handling step:

```php
if (strpos($line, ';') !== FALSE) {
    $host = parse_url($line);
    if (isset($host['host'])) {   // <- unreachable
        $line = $host['host'];
    } else {
        $line = strstr($line, ';', TRUE);
    }
}
```

The `isset($host['host'])` TRUE branch is **dead code**: `parse_url()` only populates `'host'` when its input begins with a `//` authority, but the earlier `/` truncation step (which runs BEFORE the `;` step) unconditionally keeps only the text preceding the first `/`, so any `//`-prefixed remainder is truncated to `''` before `parse_url()` is ever reached. Verified by probe: feeding `//d.com;jsessionid=x` through the helper yields `''` via the `/` truncation, never reaching `parse_url()` with a surviving `//` prefix.

This is **pre-existing** — the identical order existed in the original `if (!$lite)` loop body before the extraction; #1117/#1119 preserved it verbatim (behaviour-preserving), so it was deliberately out of scope there.

**Proposed cleanup:** collapse the `;` step to the `strstr($line, ';', TRUE)` fallback (drop the dead `parse_url`/`isset` branch), OR — if a URL with an authority *should* be handled — move the `;` step before the `/` truncation. Whichever is chosen, pin the decision with a `PfbDnsblExtractHostTest` row so the branch stops being untestable.

Low priority / cleanup.

### PR DESCRIPTION
**Pull request by andrebrait**
Fixes #1237.

## The issue's premise was wrong — and the correction changes the fix

#1237 reported the `isset($host['host'])` branch in `pfb_dnsbl_extract_host()`'s `;` step as **dead code**, reasoning that `parse_url()` only populates `host` from a `//` authority, which the earlier `/` truncation always destroys.

That reasoning is half right. The `//` route *is* dead — but `parse_url()` has a second, authority-free way to set `host`: a scheme-less `text:digits` shape, which needs no `/` at all.

```text
parse_url('d.com;jsessionid=1:80')  =>  ['host' => 'd.com;jsessionid=1', 'port' => 80]
```

So the branch is **live**, and it is actively harmful. Whenever it fires, the `;` is necessarily *inside* the returned host (a `;` after the colon makes the port non-numeric, so `parse_url` sets no host at all). A host containing `;` always fails the downstream `pfb_filter(PFB_FILTER_DOMAIN)` validation, so the line is dropped and logged as a parse failure — where the `strstr($line, ';', TRUE)` fallback would have salvaged the clean domain.

The branch could therefore never produce a usable result; it could only lose lines.

## The change

Collapse the `;` step to its `strstr(';', TRUE)` fallback:

```text
'd.com;jsessionid=1:80'   ->  'd.com'   (was: dropped as a parse failure)
'evilhost;x:99'           ->  'evilhost'
'd.com;jsessionid=x'      ->  'd.com'   (unchanged — already the fallback path)
'//d.com;jsessionid=x'    ->  ''        (unchanged — killed by the '/' truncation)
'[2604:2dc0::];x'         ->  '2604:2dc0::'  (unchanged — v6 never reached the branch)
```

This is a **behaviour change**, not the dead-code deletion the issue promised: feed lines of the `host;param:digits` family now resolve to their domain and get blocked, instead of being silently dropped. It makes the step consistent with the already-pinned `d.com;jsessionid=x` -> `d.com` behaviour.

## Tests

Red-first. Five new tests in `PfbDnsblExtractHostTest.php` (21 total, up from 16), each asserting both strict and lenient mode. Two of them failed red against untouched production code with exactly the mis-parsed values (`d.com;jsessionid=1` instead of `d.com`, `evilhost;x` instead of `evilhost`), then passed green with zero test edits. The other three pin the shapes that must *not* change — the `//` prefix, a `/` after the `;`, and a bracketed IPv6 literal carrying a `;` — so a future reordering of the pipeline steps trips them.

All 16 pre-existing tests, including both ordering-invariant tests, are byte-unmodified and green.

## ADR

ADR-62 §1.2's pipeline inventory described this step as `` `;`-`parse_url` ``. Since the mechanism is gone, the Accepted ADR's text is amended in the same change (second commit).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01BBkLQcR4cz5QCq6AVjdzuM

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved DNSBL hostname extraction by consistently truncating values at the first semicolon.
  * Prevented semicolon parameters from being misinterpreted as hostname or port data.
  * Preserved correct handling of slashes, edge-case inputs, and bracketed IPv6 addresses.

* **Tests**
  * Added coverage for semicolon truncation across strict and lenient processing modes.

### CODE REVIEW

**Comment by andrebrait** [comment]
## Review audit trail

**Reviews received:** Claude adversarial review (`review-single` workflow, `model: sonnet`, effort `xhigh` — sonnet chosen because the PR is small: 3 files, ~70 changed lines; Fable is temporarily unavailable) · **CodeRabbit** (clean: "No actionable comments were generated") · **GitHub Copilot** (1 inline finding) · **Snyk** (no status on this PR — not active here).

### Findings ledger

**1. Copilot — `pfblockerng.inc:2003`: the breadcrumb comment says "URL-authority parse", but the failure mode fires without a `//` authority.** → **APPLIED** @ `5b8cb177`.

Correct, and pointedly so. The `//`-authority route is exactly the mental model that made this issue misdiagnose the branch as dead in the first place; describing the fix's rationale in those terms would have re-seeded the same misunderstanding for the next reader. The comment now names the actual reachable mechanism — `parse_url()`'s scheme-less `host:port` heuristic, which needs no `//` at all. Comment-only change; gates re-run green (`php -l`, PHPUnit 3123/3123, PHPStan, PHPCS, markdownlint).

**2. Claude review (nitpick) — a pre-existing test's in-body comment was edited, against the "all 16 pre-existing tests byte-unmodified" acceptance criterion.** → **SKIPPED.**

The criterion's wording was mine and it was too literal. `testSemicolonWithoutParseUrlHostFallsBackToStrstr`'s comment asserted that `parse_url()` "never sets `host`" — a claim this very PR proves false in general. Leaving it would have shipped a stale comment, which CLAUDE.md's gate item 5 (comment reconciliation) forbids. The test's **assertions** are byte-identical and it is green before and after; only the prose changed. Evidence: `git diff origin/devel...HEAD -- tests/php/PfbDnsblExtractHostTest.php` shows a comment-only hunk on that test; `vendor/bin/phpunit tests/php/PfbDnsblExtractHostTest.php` → OK (21 tests, 51 assertions).

**3. Claude review (nitpick) — `scripts/agent/verify-red-proof.sh` hardcodes `HEAD~1`, so a bare re-run against this branch (fix commit + stacked doc-only ADR commit) misreports `RED-FAIL`.** → **DEFERRED** → #1249.

Real, and pre-existing tooling — orthogonal to this change, so it gets its own issue rather than riding along. This PR's red→green proof genuinely holds against the true baseline (reverting `pfblockerng.inc` to `origin/devel` produces exactly the two expected failures with the mis-parsed values; restoring `HEAD` gives 21/21 green, test file unedited between red and green, `hash-object f3291e34…`).

### Coverage-matrix re-tick after the APPLY

The applied fix is comment-only inside the already-covered `;` step — no new behaviour, no matrix row added or invalidated. All seven rows from the step brief remain ticked; the 16 pre-existing tests (including both ordering-invariant tests) stay green and assertion-identical.

The Claude reviewer independently brute-forced 500+ crafted `;`-bearing inputs (the port-like-suffix family, bare and bracketed IPv6, metacharacters, tabs, raw UTF-8/IDN labels, invalid UTF-8, oversized hosts, leading/trailing `;`, double colons) and found no shape where the removed `parse_url` path produced a valid host that plain truncation fails to reproduce — every divergent case leaves a `;` in the old output, which `pfb_filter(PFB_FILTER_DOMAIN)` rejects unconditionally. That is the load-bearing claim behind calling this branch strictly harmful.

---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

File: src/usr/local/pkg/pfblockerng/pfblockerng.inc [hunk 1]
```diff
@@ -1999,14 +1999,10 @@ function pfb_dnsbl_extract_host(string $line, bool $strict, string $header, stri
 		$line = strstr($line, '?', TRUE);
 	}
 
-	// If special characters found, parse line for host
+	// issue #1237: a URL-authority parse used to mis-detect 'host;param:digits' as a
+	// hostname, keeping the ';param' remainder -- always truncate at the first ';' instead.
```
[the remark this answers is not in the document]
Comment (andrebrait) [reply src/usr/local/pkg/pfblockerng/pfblockerng.inc]: Applied in 5b8cb177 — good catch, and a pointed one.

The `//`-authority framing is precisely the mental model that made issue #1237 misdiagnose this branch as dead code, so leaving it in the comment would have re-seeded the same misunderstanding for the next reader. The comment now names the mechanism that actually fires: `parse_url()`'s scheme-less `host:port` heuristic, which needs no `//` at all.
